### PR TITLE
Updating the  doc for maxAutoLockRenewDuration setter in all receiver and processor builders, timeout argument for sync-receive and removing dead code in v1 path

### PR DESCRIPTION
**Pull request by anuchandy**

### CODE REVIEW

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 1]
```diff
@@ -1566,11 +1566,23 @@ private ServiceBusSessionProcessorClientBuilder() {
          * Sets the amount of time to continue auto-renewing the lock. Setting {@link Duration#ZERO} or {@code null}
          * disables auto-renewal. For {@link ServiceBusReceiveMode#RECEIVE_AND_DELETE RECEIVE_AND_DELETE} mode,
          * auto-renewal is disabled.
+         * <p>
+         * A Service Bus queue or subscription in a topic will have a lock duration set in the resource level.
+         * When the processor client connect to a session in the resource, the broker will apply an initial
+         * lock to the session. This initial lock only lasts for the lock duration set in the resource level.
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:1572]: "This initial lock lasts for the lock duration set at the resource level."

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 2]
```diff
@@ -1566,11 +1566,23 @@ private ServiceBusSessionProcessorClientBuilder() {
...
+         * If the client does not renew the initial lock before it expires then the session will be released and become
+         * available for other receivers. Each time the client renews the lock, the broker will extend the lock for the
+         * lock duration set in the resource level. It means, to keep the session locked for a long time, the client may
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:1575]: I think "at the resource level" is better than "in the resource level"?  but that might be me.

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java [hunk 3]
```diff
@@ -1566,11 +1566,23 @@ private ServiceBusSessionProcessorClientBuilder() {
...
+         * have to renew the lock multiple times over multiple network calls. Client runs a local task that attempts
+         * to renew the lock each time before it expires. The {@code maxAutoLockRenewDuration} configures how long this
+         * local task should be running, hence it is possible that last renewed lock may be valid for some time after
+         * the task disposal.
+         * </p>
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java:1580]: "To keep the session locked, the client will have to continuously renew the session lock before its expiration.  {@code maxAutoLockRenewDuration} controls how long the background renewal task runs.  So, it is possible that the previous renewed lock can be valid after the renewal task is disposed."

**Review by conniey — APPROVED** [review]
Thanks!

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java (hunk 2 above)
Comment (anuchandy) [reply sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusClientBuilder.java]: agree, reworded

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java [hunk 4]
```diff
@@ -492,7 +494,16 @@ IterableStream<ServiceBusReceivedMessage> peekMessages(int maxMessages, long seq
      * Receives an iterable stream of {@link ServiceBusReceivedMessage messages} from the Service Bus entity. The
      * receive operation will wait for a default 1 minute for receiving a message before it times out. You can
      * override it by using {@link #receiveMessages(int, Duration)}.
-     *
+     * <p>
+     * The 1-minute timeout is a client-side feature. Each time the application calls receiveMessages, a timer is started
+     * on the client that when expires will terminate the IterableStream returned from this method. Timeout being
+     * a client-side feature means it is impossible to cancel any messages request that already made it to the broker.
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java:500]: "..cancel any message requests"

File: sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java [hunk 5]
```diff
@@ -492,7 +494,16 @@ IterableStream<ServiceBusReceivedMessage> peekMessages(int maxMessages, long seq
      * Receives an iterable stream of {@link ServiceBusReceivedMessage messages} from the Service Bus entity. The
      * receive operation will wait for a default 1 minute for receiving a message before it times out. You can
      * override it by using {@link #receiveMessages(int, Duration)}.
-     *
+     * <p>
+     * The 1-minute timeout is a client-side feature. Each time the application calls receiveMessages, a timer is started
```
Comment (conniey) [inline sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/ServiceBusReceiverClient.java:498]: personal: when I am mentioning code invocations, I try to use `{@code receiveMessages}` or `{@link }` so users know it's a method or something they can call.

**Review by lmolkova — APPROVED** [review]

**Review by conniey — APPROVED** [review]

**Pull request merged**